packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1049,18 +1085,24 @@ export class TransactionController extends BaseController<
    * and emitting a `<tx.id>:finished` hub event.
    *
    * @param transactionID - The ID of the transaction to cancel.
+   * @param actionId - Unique ID to prevent duplicate requests
    */
-  private cancelTransaction(transactionID: string) {
-    const transactionMeta = this.state.transactions.find(
-      ({ id }) => id === transactionID,
-    );
+  private cancelTransaction(transactionID: string, actionId?: string) {
+    const transactionMeta = this.state.transactions.find((transaction) => {
+      if (actionId) {
```

Comment:
Is this check required? Won't the transaction ID still be the same if it's a duplicate call as we've got the same transaction meta thus the same ID when the code reaches here?

Reply:
I'm wondering if you meant to make this change to `stopTransaction` which confusingly is equivalent to `createCancelTransaction` in the extension?

In core at least, `cancelTransaction` is reject an unapproved transaction, and `stopTransaction` is cancel an already submitted transaction.

Reply:
I was reviewing my changes in the `cancelTransaction` after reading your clarification and the equivalence between the extension got me confused and misled me with this additional change. I reverted this part.